modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexRowDeserializer.java

```diff
@@ -27,5 +27,5 @@
      * @param indexRow Index row.
      * @return Values of the indexed columns.
      */
-    Object[] indexedColumnValues(IndexRow indexRow);
+    IndexRow row(IndexBinaryRow indexRow);
```

Comment:
The interface looks weird.
Why do we ever need to deserialize a whole index row?

Reply:
Not catch an idea. What do you think should be deserialized? 
`IndexRow` is an interface. Any deserializations may be implemented lazy inside the interface implementation.